script/lib/createBuildDescriptor.js

```diff
@@ -0,0 +1,35 @@
+const fs = require("fs");
+const { exec } = require("child_process");
+
+function createBuildDescriptor(config) {
+    let yarnLockContent = fs.readFileSync("yarn.lock");
+
+    let dir = "dist/";
+
+    if (config["dirs"] && config["dirs"]["dist"]) {
+        dir = config["dirs"]["dist"];
+        if (!dir.endsWith("/")) dir += "/";
+    }
+
+    fs.writeFile(dir + "yarn.lock", yarnLockContent, err => {
+        if (err) throw err;
+        console.log("yarn.lock descriptor has been created");
+    })
+
+    exec("git describe --abbrev=7 --tags --dirty=+dirty --always", (err, stdout, stderr) => {
+        if (err) {
+            console.error(err);
+            fs.writeFile(dir + "version.txt", stderr, err => {
```

Comment:
@titkov49 Just a question - I thought the file with version should have been a `json` file or `md`, so why `txt` ?

Reply:
It is a single line of text, you can read that from JavaScript ... so it is fine as a .txt

BTW- we should also create "About" page in the asab-webui that should display the version info to the user (from user drop down).

Reply:
Noted and ticket created